harness/tests/core/test_checkpoint.py

```diff
@@ -141,3 +143,70 @@ def do_test() -> None:
                     pass
             storage_manager.restore_path.assert_called_once()
             storage_manager.restore_path.reset_mock()
+
+
+@pytest.mark.parametrize(
+    "resources,expected_merged,expected_conflicts",
+    [
+        ([{"file0": 0}, {"file1": 0}], {"file0": 0, "file1": 0}, {}),
+        ([{"file0": 0}, {"file0": 0}], {"file0": 0}, {"file0": [0, 1]}),
+        ([{"dir1/": 0}, {"dir1/": 0}], {"dir1/": 0}, {}),
+        ([{"file1/": 0}, {"file1": 0}], {"file1/": 0, "file1": 0}, {"file1": [0, 1]}),
+        ([{"dir1/file1": 0}, {"file1": 0}], {"dir1/file1": 0, "file1": 0}, {}),
+        (
+            [{"dir1/file1": 0}, {"dir1/file1/": 0}],
+            {"dir1/file1": 0, "dir1/file1/": 0},
+            {"dir1/file1": [0, 1]},
+        ),
+    ],
+)
+def test_merge_files(
+    resources: List[Dict[str, int]],
+    expected_merged: Dict[str, int],
+    expected_conflicts: Dict[str, List[int]],
+) -> None:
+    merged, conflicts = core._checkpoint.merge_resources(resources)
+    assert conflicts == expected_conflicts
+    assert merged == expected_merged
+
+
+@pytest.mark.parametrize(
+    "metadata,expected_merged,expected_conflicts",
+    [
+        ([{"key1": 0}, {"key2": 0}], {"key1": 0, "key2": 0}, {}),
```

Comment:
I suggest using letters instead of `key1`, `key2`.  I think it's way easier to read afterwards.